Objects/codeobject.c

```diff
@@ -1797,9 +1797,9 @@ code_richcompare(PyObject *self, PyObject *other, int op)
     for (int i = 0; i < Py_SIZE(co); i++) {
         _Py_CODEUNIT co_instr = _PyCode_CODE(co)[i];
         _Py_CODEUNIT cp_instr = _PyCode_CODE(cp)[i];
-        uint8_t co_code = co_instr.op.code;
+        uint8_t co_code = _Py_GetBaseOpcode(co, i);
```

Comment:
I don't think that call this API at this moment is correct.
Because it will call _PyOpcode_Deopt twice in the end.

Reply:
Or move co_code/cp_code = _PyOpcode_Deopt[cp_code]; only if cp_code or co_code is ENTER_EXECUTOR

Reply:
FYI, If we decide to fix `_Py_GetBaseOpcode()`, we can update https://github.com/python/cpython/issues/107265 too, which was decided not to fix :)

Reply:
And maybe we can close https://github.com/python/cpython/issues/107265 in one short.

Reply:
IIRC there are also cases where we absolutely want the actual opcode, so in those cases we still need to handle ENTER_EXECUTOR explicitly. But yeah, fixing this in `_Py_GetBaseOpcode()` might reduce the special cases elsewhere. In a new PR, please. :-)

Reply:
**IMPORTANT!** Do not attempt to "fix" `_Py_GetBaseOpcode()` by adding `ENTER_EXECUTOR` handling. This will leave the `oparg` incorrect. That's why that bullet in gh-107265 uses ~strikethrough~.